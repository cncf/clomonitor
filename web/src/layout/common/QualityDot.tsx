import styles from './QualityDot.module.css';

interface Props {
  level: number;
  className?: string;
}

const QualityDot = (props: Props) => {
  return (
    <div
      data-testid="quality-dot"
      className={`me-2 border border-1 rounded-pill position-relative ${styles.quality} ${
        styles[`level${props.level}`]
      } ${props.className}`}
    />
  );
};

export default QualityDot;
