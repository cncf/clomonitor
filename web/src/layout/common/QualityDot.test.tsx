import { render, screen } from '@testing-library/react';

import QualityDot from './QualityDot';

describe('QualityDot', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<QualityDot level={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<QualityDot level={1} />);

    expect(screen.getByTestId('quality-dot')).toBeInTheDocument();
    expect(screen.getByTestId('quality-dot')).toHaveClass('level1');
  });

  it('renders level 3 dot', () => {
    render(<QualityDot level={3} />);

    expect(screen.getByTestId('quality-dot')).toHaveClass('level3');
  });
});
