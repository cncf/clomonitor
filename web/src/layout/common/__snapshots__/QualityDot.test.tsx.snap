// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QualityDot creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="me-2 border border-1 rounded-pill position-relative quality level1 undefined"
    data-testid="quality-dot"
  />
</DocumentFragment>
`;
